### ISSUE TITLE
How to revert Google Drive synced backup to older version?

### ISSUE THREAD

**Issue by chaoscreater** [violentmonkey/violentmonkey#2024]
As per another Github issue, the data backed up to Google Drive is not visible to the user. I can't find the file and can't look at the version history. I'm trying to revert the Google Drive synced data to an older version. The reason is that I have 2 machines and both synced to the same Google Drive account. One machine had old scripts and synced to Google Drive and that in turn downloaded onto my other machine and now I have old copies of the script, which is annoying. I know that prior to the latest sync, it contained the data I want and I have no way to find it and restore it.

**Comment by gera2ld** [comment]
If sync is already done, I don't think you can get back the overridden data.
If you have the desired data locally, you can import it and let it sync again.
BTW you can find the removed scripts during sync in Recycle Bin.

**Comment by chaoscreater** [comment]
> If sync is already done, I don't think you can get back the overridden data. If you have the desired data locally, you can import it and let it sync again. BTW you can find the removed scripts during sync in Recycle Bin.

Sync was already done.

Both browsers have the same list of scripts prior to the sync. One of the scripts got updated in Browser A but browser B didn't sync the update, because sync was broken. After I noticed the issue and repaired the sync, Browser B then overwrote the sync from Browser A. Since the sync was never broken in Browser A, it meant that the new sync from Browser B is the latest, so all changes in Browser A got replaced by the changes in Browser B.

There's no data stored locally. That's the point of using the sync to a cloud drive.

No scripts were deleted. They were simply updated from old version to new version. However, because of the broken sync, it went from new version back to old version.

**Comment by gera2ld** [comment]
> I read somewhere that the data stored in Google Drive is application data and not visible to the user.

Yes, this is recommended by Google so that we don't have access to your other files. It's Google who chooses to hide the files from users.

> Is this the case for other drives like Dropbox?

We also store scripts in application data of Dropbox, but it seems accessible to users.

**Comment by chaoscreater** [comment]
I've just tested syncing to Dropbox and looks like it does show files to the users, which is exactly what I want. Now I get to use version history in case I need to rollback due to a bad sync. This is great. Ideally, ViolentMonkey should support sync version history rollback if possible, but this will do for now.

![image](https://github.com/violentmonkey/violentmonkey/assets/18227319/6f66d6d5-c215-472c-95d2-9ab7d65b6575)

**Comment by gera2ld** [comment]
> Data will sync the first time, but not afterwards.

Is it because of wrong sync direction, i.e. always from Dropbox to local?
